Testing Page HTML and CSS Layout

chrisn02 opened the pull request:
Have created the HTML and CSS layout for the testing page. The subject and test names are displayed in the top corner of the screen. There is space for the question to go with the question number as well as a progress bar underneath it. The answer boxes increase in size slightly when hovered over and while it has not been implemented yet I am planning on having it so that when you click on an answer a button will appear on the bottom right corner of the page to advance to the next question. As seen in the image there is some blank space but this was not there when I was originally making the page, so will have to adjust based on screen size.
![TestingPage](https://github.com/Jforjo/CSD/assets/59711723/9483e7f8-8999-4806-ae08-cac1b0564f1d)

chrisn02 committed:
Testing page started

chrisn02 committed:
Added slightly more to page

Just working on basic HTML structure so far

chrisn02 committed:
Created the answer boxes

Created the answer boxes that the user will click on. Very basic design so far

chrisn02 committed:
Added question textbox area

chrisn02 committed:
Changed position of answer boxes

chrisn02 committed:
Added some more styling

Added some more styling to the testing page, such as changed the background colour to black. All styling is currently part of the testing-page rather than the CSS file for now

chrisn02 committed:
Navbar styling

chrisn02 committed:
Change file extension to HTML (temp)

chrisn02 committed:
Made contents on page slightly bigger

chrisn02 committed:
Added space for the progress bar

chrisn02 committed:
Moved styling to separate CSS file

also renamed file type from html back to php

chrisn02 committed:
Have repositioned contents of page slightly

Needs to be done better though, need to fix the way it looks when on a different size screen. May redo completely how page looks

chrisn02 committed:
Set fixed height and width for question box

Done this so that it is always the same size no matter the length of the question

Also changed positioning of progress bar

chrisn02 committed:
Redone positioning for when on different size screen

Might need to add more for different sizes again

chrisn02 committed:
Redone positioning of answer boxes and progress bar

chrisn02 committed:
Resized the question text box

Only done because the questions aren't likely to be as long as the placeholder text I was using, which as a result will cause blank space on the page. Will revert back if needed though. Have also added a temporary border to it so we have a visual on where exactly the question text area is

chrisn02 committed:
Added some design to the answer boxes

gchodancar19 reviewed (approved):
Is there anything else to added like coding.

Jforjo commented on style.css:
```diff
@@ -0,0 +1,298 @@
+        grid-template-columns: 1fr 1fr;
+        gap: 25px;
+        width: 75%;
+        height: 20%;
```
The answers box should also be taller, as I think each box is too small in height right now.

Jforjo commented on style.css:
```diff
@@ -0,0 +1,298 @@
+    .answers
+    {
+        display: grid;
+        grid-template-columns: 1fr 1fr;
```
When the screen gets too small you should just use 1 column so the questions stack on top of each other.

Jforjo commented on style.css:
```diff
@@ -0,0 +1,298 @@
+        max-width: 900px;
+        min-width: 900px;
+        max-height: 100px;
+        min-height: 100px;
```
There is no need to have both `max-height` and `min-height` - just use `height`.

Jforjo commented on style.css:
```diff
@@ -0,0 +1,298 @@
+        text-align: center;
+        margin-bottom: 20px;
+        max-width: 900px;
+        min-width: 900px;
```
This `min-width` will cause the question to spill off the screen when the view-width gets too small. You can just remove this.

Jforjo commented on style.css:
```diff
@@ -0,0 +1,298 @@
+        justify-content: center;
+        align-items: center;
+        text-align: center;
+        margin-bottom: 20px;
```
This is redundant as you have `margin: 0 auto;` below, which overwrites this `margin-bottom`.

Jforjo reviewed (changes requested):
See individual line comments.
Also, you could try to get the progress bar working by putting a `<i>` tag in with `inset: 0` and a percentage width. This percentage width being the amount the progress bar is filled.

chrisn02 commented:
Thanks for your feedback, I will work on this now.

chrisn02 committed:
Have done requested changes

Have the changes that were requested in pull request #5

Includes things like the progress bar having a bar that changes based off percentage.

chrisn02 commented:
Have made the requested changes. Have made the answer boxes taller and when on a smaller screen such as mobile, the boxes will stack on top of each other. The progress bar now fills based on percentage, the image is on 75% as an example. Let me you what you think.
![TestingPage_Fix](https://github.com/Jforjo/CSD/assets/59711723/5baf1f38-2448-455e-9c78-7b3df982bb0c)
![TestingPage_MobileView](https://github.com/Jforjo/CSD/assets/59711723/cef33ced-13d4-4924-aecd-f9943173b71d)

Jforjo reviewed (approved):
Looks good, but should should also make sure it looks good if there are only 2 or 3 answers too.

chrisn02 committed:
Merge branch 'sudo-main' into CN-TestingPage

Jforjo reviewed (approved):
I don't think you can merge something if you've also made changes, since we have to approve it again. You've added a new line with nothing on it, which is a change that we need to review. When merging make sure there are no changes next time.

chrisn02 commented:
I think that must've happened as I was resolving a merge conflict, apologies
